src/hivpy/hiv_status.py

```diff
@@ -24,17 +39,54 @@ def initial_HIV_status(self, population: pd.DataFrame):
         hiv_status = self._prob_HIV_initial(population[col.AGE]) > rng.random(len(population))
         return hiv_status
 
-    def update_HIV_status(self, population: pd.DataFrame):
+    def update_partner_risk_vectors(self, population):
+        """calculate the risk factor associated with each sex and age group"""
+        # Should we be using for loops here or can we do better? 
```

Comment:
So I thought about this using a `groupby` but I don't know if it can be done because we need to access HIV status and  number of partners, but we don't want to group by these, only by sex and age group. I'm not sure how we can access these additional fields if we use a group by